### PR TITLE
Reenable stackdriver logging for upgrade job now that Kyma 1.15.1 is …

### PR DESCRIPTION
**Pull request by nachtmaar**
**Description**

Changes proposed in this pull request:

- Use stackdriver logging integration which is only supported since Kyma 1.15.1 (see issue for more description)

A proof that it works can be found in this [PR with pjtester](https://github.com/kyma-project/test-infra/pull/2722/files)
Also see the job lobs in the screenshot for this pipeline: https://status.build.kyma-project.io/log?job=nachtmaar_test_of_prowjob_pre-master-kyma-upgrade-gardener-azure&id=1313069757024440320

Screenshot 2020-10-05 at 16 15 11

**Related issue(s)**

https://github.com/kyma-project/test-infra/issues/2674

### CODE REVIEW

File: prow/scripts/cluster-integration/kyma-upgrade-gardener-azure.sh [hunk 1]
```diff
@@ -390,10 +396,6 @@ fi
 createTestResources
 
 upgradeKyma
-if [[ "$?" -ne 0 ]]; then
```
Comment (sayanh) [inline prow/scripts/cluster-integration/kyma-upgrade-gardener-azure.sh:393]: Why did you remove this?

Comment (nachtmaar) [reply prow/scripts/cluster-integration/kyma-upgrade-gardener-azure.sh:393]: Good point. I forgot to commit one change which makes it clear why this error check can be removed.


Some context:

Previously the stackdriver installation overrides were created only during the upgrade (there was a change in installation overrides, therefore it was not possible to pass the overrides to Kyma 1.15.1, but for current master - and upgrade takes master). So the exit code of the prepare_stackdriver_logging was checked and returned. In case of an error (return 1), the process was exited with 1.

Comment (nachtmaar) [reply prow/scripts/cluster-integration/kyma-upgrade-gardener-azure.sh]: As discussed: we will have leave it in case the check installation step throws an error.

**Review by sayanh — APPROVED** [review]
